src/runner.rs

```diff
@@ -112,9 +115,12 @@ impl Runner for FirefoxRunner {
         command
             .env("MOZ_NO_REMOTE", "1")
             .env("NO_EM_RESTART", "1")
-            .args(&self.args[..])
-            .arg("-profile").arg(&self.profile.path)
-            .stdout(Stdio::inherit())
+            .args(&self.args[..]);
+
+        if !self.args.iter().any(|x| is_profile_arg(x)) {
```

Comment:
Aha, perhaps there is some type coercion going on here that it isn’t able to figure out if you pass a function by reference.